homeassistant/helpers/event.py

```diff
@@ -1362,15 +1363,21 @@ def async_call_later(
     delay: float | timedelta,
```

Comment:
These are meant to be as cheap as possible.

I might make a new `async_call_later_unless_stopping` for this use case to reduce the complexity/branching

Reply:
We could also somehow mark the `TimerHandle` as safe to cancel when stopping and cancel all the marked ones at stop which would be really cheap

Reply:
This would probably work. But maybe there is a better solution:

`setattr(timer_handle,'_cancel_on_hass_stop',True) `
and than `async_stop` would look at all of them and cancel them

Reply:
I couldn't set an attribute on the `timer_handle`, but I did manage to add an attribute to `HassJob`
What do you think?

Reply:
Almost everything we likely want to cancel at shutdown is likely a HassJob so that should work. It's relatively cheap so it shouldn't cause us performance pain later